### PR TITLE
EPMKHRDMH-377 [sidebar] add sidebar functionality

### PR DESCRIPTION
**Pull request by 8bitmousepunk**

### CODE REVIEW

File: www/modules/calendar/views/calendar.client.view.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="main-calendar" data-ng-controller="CalendarController as calendar">
     <header class="main-header-mobile main-header-mobile--fixed">
         <div class="main-header-mobile__main-tab-info">
-            <span class="main-header-mobile-control__icon icon icon-dehaze"></span>
+            <span class="main-header-mobile-control__icon icon icon-dehaze" ng-click="calendar.toggleSidebar()"></span>
```
Comment (ghost) [inline www/modules/calendar/views/calendar.client.view.html]: Please, add `data-` prefix to `ng`

File: www/modules/sidebar/controllers/sidebar.client.controller.es6 [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+    this.$rootScope.$on('toggle_sidebar',() => {
+      this.toggleSidebar();
+    });
+
+    this.$rootScope.$on('disable_sidebar', () => {
+      this.disableSidebar();
+    });
+
+    this.$rootScope.$on('enable_sidebar', () => {
+      this.enableSidebar();
```
Comment (ghost) [inline www/modules/sidebar/controllers/sidebar.client.controller.es6:16]: Seems we don't use `enable_sidebar`

File: www/modules/sidebar/views/sidebar.client.view.html [hunk 3]
```diff
@@ -1,8 +1,11 @@
-<div class="sidebar">
+<div ng-class="{'content-overflow': true, 'content-overflow--active': sidebarCtrl.active}" ng-click="sidebarCtrl.disableSidebar()"></div>
+<div ng-class="{'sidebar-active': sidebarCtrl.active, sidebar: true}">
```
Comment (ghost) [inline www/modules/sidebar/views/sidebar.client.view.html]: Please, add `data-` prefix to `ng`

Comment (ghost) [reply www/modules/sidebar/views/sidebar.client.view.html]: Seems classes `sidebar ` ,` content-overflow `we can take out from expression

File: www/modules/sidebar/views/sidebar.client.view.html [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
     <header>
         <div class="user-info">
             <div class="user-info__avatar">
-                <img data-ng-src="{{user.profileImageURL}}" alt="{{user.username}}">
+                <a ui-sref="restricted.profile" ng-click="sidebarCtrl.disableSidebar()">
```
Comment (ghost) [inline www/modules/sidebar/views/sidebar.client.view.html]: Please, add `data-` prefix

File: www/modules/sidebar/views/sidebar.client.view.html [hunk 5]
```diff
@@ -12,33 +15,33 @@
...
                 <img src="modules/sidebar/img/event.png">
-                <a ui-sref="restricted.todo_state.events.list">Events</a>
+                <a ui-sref="restricted.todo_state.events.list" ng-click="sidebarCtrl.disableSidebar()">Events</a>
             </li>
             <li class="separator"></li>
             <li>
                 <img src="modules/sidebar/img/repeat.png">
-                <a ui-sref="restricted.migrate">Import content</a>
+                <a ui-sref="restricted.migrate" ng-click="sidebarCtrl.disableSidebar()">Import content</a>
             </li>
```
Comment (ghost) [inline www/modules/sidebar/views/sidebar.client.view.html]: Please, add `data- `prefix for all links here

**Review by ghost — CHANGES_REQUESTED** [review]

File: www/modules/sidebar/controllers/sidebar.client.controller.es6 (hunk 2 above)
Comment (8bitmousepunk) [reply www/modules/sidebar/controllers/sidebar.client.controller.es6:16]: It is for future iterations.

**Pull request merged**